editors/vscode/src/common.ts

```diff
@@ -217,13 +218,83 @@ export function activate(
         new wasm_preview.PreviewSerializer(context),
     );
 
-    vscode.workspace.onDidChangeTextDocument(async (ev) => {
+    let startupLogged = false;
+    let maybeSendStartupTelemtryEvent = async () => {
+        if (startupLogged) {
+            return;
+        }
+
         if (
-            ev.document.languageId !== "slint" &&
-            ev.document.languageId !== "rust"
+            vscode.workspace.textDocuments.some((doc) => {
+                if (doc.languageId === "slint") {
+                    return true;
+                }
+                if (doc.languageId === "rust") {
+                    if (doc.getText().includes("slint::slint!")) {
```

Comment:
```suggestion
                    if (doc.getText().includes("slint!")) {
```
in case people `use` that macro

Reply:
Technically someone could also write `slint !`... but that is getting a bit esoteric I think :-)